crates/jit/src/instantiate.rs

```diff
@@ -99,10 +99,15 @@ impl CompilationArtifacts {
     /// The `use_paged_init` argument controls whether or not an attempt is made to
     /// organize linear memory initialization data as entire pages or to leave
     /// the memory initialization data as individual segments.
+    ///
+    /// The `prefer_parallel_compilation` is used to control whether compilation should be performed
+    /// using multiple threads or not. This has only an effect if the feature `parallel-compilation`
+    /// is enabled.
     pub fn build(
         compiler: &Compiler,
         data: &[u8],
         use_paged_mem_init: bool,
+        #[cfg(feature = "parallel-compilation")] parallel_compilation: bool,
```

Comment:
Could this argument be baked into the `Compiler` itself rather than passed as a parameter? I also think it's fine for it to be an unconditional parameter to avoid the `#[cfg]` on usage